docs/proposals/odo-v3-cli.md

```diff
@@ -0,0 +1,305 @@
+To deploy your component to a cluster use “odo deploy”.   
+```
+
+Proof of concept was implemented in https://github.com/kadel/odo-v3-prototype/blob/main/cmd/init.go
+
+
+
+#### Flags
+- `--name` - name of the component (required)
+- `--devfile` - name of the devfile in devfile registry (required if `--devfile-path` is not defined)
+- `--devfile-registry` - name of the devfile registry (as configured in `odo registry`). It can be used in combination with `--devfile`, but not with `--devfile-path` (optional)
+- `--starter` - name of the  starter project (optional)
+- `--devfile-path` - path to a devfile. This is alternative to using devfile from Devfile registry. It can be local filesystem path or http(s) URL (required if `--devfile` is not defined)
+
+
+If no flag is specified it should enter interactive mode.
+If even a single optional flag is specified then run in non-interactive mode and requires all required flags.
+
+
+#### Command behavior and error states
+
+- the result of running `odo init` command should be local devfile.yaml saved in the current directory, and starter project extracted in the current directory (if user picked one)
+- running `odo init` in non-empty directory exits with error
+  
+  ```
+  The current directory is not empty. You can bootstrap new component only in empty directory.
+  If you have existing code that you want to deploy use `odo deploy` or use `odo dev` command to quickly iterate on your component.
+  ```
+
+- command should use registry as configured in `odo registry` command. If there is multiple registries configured it should use all of them.
+
+
+
+
+### `odo deploy`
+
+Deploying application (outer loop)
+[#5298](https://github.com/redhat-developer/odo/issues/5298)
+
+#### Interactive mode
+#### Flags
+#### Command behavior and error states
+
+### `odo build-images`
+mostly as it is in v2
+
+### `odo dev`
+
+Running the application on the cluster for **development** (inner loop)
+
+[#5299](https://github.com/redhat-developer/odo/issues/5298)
+
+
+### `odo list`
+
+Listing "stuff" created by odo.
+
+- without arguments lists components in local Devfile and in cluster
+- `odo list services` list services defined in local devfile and in cluster
+- `odo list endpoints` list endpoints defined in local devfile and in cluster
+- `odo list bindings` list bindings defined in local devfile and in cluster 
+
+
+### `odo preference`
+
+Configures odo behavior like timeouts.
+- UpdateNotification  - keep
+- NamePrefix          - remove
+- Timeout             - add explanations  
+- BuildTimeout        - remove
+- PushTimeout         - add explanations
+- Ephemeral           - keep
+- ConsentTelemetry    - keep 
+
+
+
+### `odo project`
+mostly as it is
+
+### `odo registry`
+mostly as it is just drop  support "github"  registry
+
+### `odo describe`
+TODO
+
+### `odo exec`
+mostly as it is
+
+### `odo status`
+mostly as it is
+
+## Commands that will be added in v3.0.0-alpha2
+
+### `odo create`
```

Comment:
What will happen to the v2 `odo create` command? Will it remain as is, or removed in favour of `odo init` / `odo dev`?

Reply:
`odo create` on itself won't be a valid command. But it will have subcommand to create other "things" like `odo create service`